WalletWasabi/Crypto/CryptoGuards.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using WalletWasabi.Crypto.Groups;
+
+namespace WalletWasabi.Crypto
+{
+	public static class CryptoGuard
+	{
+		[DebuggerStepThroughAttribute]
+		public static GroupElement NotInfinity(string parameterName, GroupElement groupElement)
+			=> groupElement.IsInfinity switch
+			{
+				true => throw new ArgumentException("Point at infinity is not a valid value.", parameterName),
+				false => groupElement
+			};
+
+		[DebuggerStepThroughAttribute]
+		public static IEnumerable<GroupElement> NotInfinity(string parameterName, IEnumerable<GroupElement> groupElements)
```

Comment:
if guarding a `GroupElementsVector`, will this preserve the type on assignment to a variable/field of the concrete type?

Reply:
Yes.